api/src/main/java/org/pentaho/platform/api/genericfile/IGenericFileService.java

```diff
@@ -291,11 +292,64 @@ default IGenericFile getFile( @NonNull String path ) throws OperationFailedExcep
+
+  /**
+   * Permanently deletes a file, given its path.
+   *
+   * @param path The file path to be permanently deleted. This path must correspond to a file that is in the trash/deleted.
    * @throws AccessControlException   If the current user cannot perform this operation.
+   * @throws InvalidPathException     If the specified path does not correspond to a file that are in the trash/deleted.
+   * @throws NotFoundException        If the specified path does not exist, or the current user is not allowed to access it.
    * @throws OperationFailedException If the operation fails for some other (checked) reason.
    */
-  void deleteFilesPermanently( @NonNull List<GenericFilePath> paths ) throws OperationFailedException;
+  void deleteFilePermanently( @NonNull GenericFilePath path ) throws OperationFailedException;
+
+  /**
+   * Deletes files by sending them to the trash, given their paths.
+   *
+   * @param paths The list of file paths to be deleted.
+   * @throws AccessControlException        If the current user cannot perform this operation.
+   * @throws BatchOperationFailedException If the batch operation fails for some reason.
+   * @throws OperationFailedException      If the operation fails for some other (checked) reason.
+   */
+  void deleteFiles( @NonNull List<GenericFilePath> paths ) throws OperationFailedException;
+
+  /**
+   * Deletes a file by sending it to the trash, given its path.
+   *
+   * @param path The file path to be deleted.
+   * @throws AccessControlException   If the current user cannot perform this operation.
+   * @throws NotFoundException        If the specified path does not exist, or the current user is not allowed to access it.
+   * @throws OperationFailedException If the operation fails for some other (checked) reason.
+   */
+  void deleteFile( @NonNull GenericFilePath path ) throws OperationFailedException;
+
+  /**
+   * Restores files, given their paths.
+   *
+   * @param paths The list of file paths to be restored. These paths must correspond to files that are in the trash/deleted.
+   * @throws AccessControlException        If the current user cannot perform this operation.
+   * @throws BatchOperationFailedException If the batch operation fails for some reason.
+   * @throws OperationFailedException      If the operation fails for some other (checked) reason.
+   */
+  void restoreFiles( @NonNull List<GenericFilePath> paths ) throws OperationFailedException;
+
+  /**
+   * Restores a file, given its path.
+   *
+   * @param path The file path to be restored. This path must correspond to a file that is in the trash/deleted.
+   * @throws AccessControlException   If the current user cannot perform this operation.
+   * @throws InvalidPathException     If the specified path does not correspond to a file that are in the trash/deleted.
+   * @throws NotFoundException        If the specified path does not exist, or the current user is not allowed to access it.
```

Comment:
Idem:

`InvalidPathException` is typically tied to the syntax of the path.

As said in other comments, if given a path which is not in the trash, a `NotFoundException` would be thrown -- it is not found in the trash!